GenericTensor/Functions/Composition.cs

```diff
@@ -86,5 +86,22 @@ internal static class Composition<T, TWrapper> where TWrapper : struct, IOperati
                 return res;
             }
         }
+        
+        public static void Fold<TCollapse, U, UWrapper>(TCollapse collapse, GenTensor<U, UWrapper> acc, GenTensor<T, TWrapper> t, int axis)
+            where TCollapse : struct, HonkPerf.NET.Core.IValueDelegate<U, T, U>
+            where UWrapper : struct, IOperations<U>
+        {
+            for (int i = axis; i > 0; i--)
+                t.Transpose(i, i - 1);
```

Comment:
```suggestion
                t.Transpose(i, i - 1); // Move the axis we want to reduce to the front for GetSubtensor. Order is important since it is directly reflected in the output shape.
```